### PR TITLE
fix: set sites value if pin map id is empty in Get Connections Impl.vi

### PR DESCRIPTION
**Pull request by varshini-senthil**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-labview/blob/main/CONTRIBUTING.md). _(Required)_
- [ ]  Automatically post PR comments with images for G code changes? _(Recommended for small changes)_

### What does this Pull Request accomplish?

- Updated logic to set the `pin map context.sites` value to [0] if the array is empty and the `pin map context.pin_map_id` value is also empty string.

### Why should this Pull Request be merged?

- To fix the issue mentioned in the [PR](https://github.com/ni/measurement-plugin-labview/pull/598).

### What testing has been done?

- Tested by running the client and verified if the `pin map context.sites` value is set to [0] for empty `pin_map_id`.

### CODE REVIEW

**Comment by varshini-senthil** [comment]
`Get Connections Impl.vi`
Diff image
Before:

image

After:

image

**Review by dixonjoel — APPROVED** [review]

File: Source/Runtime/Clients/Session Management V1/Session Reservation/Get Connections Impl.vi
Comment (jasonmreding) [inline Source/Runtime/Clients/Session Management V1/Session Reservation/Get Connections Impl.vi:1]: I might just always ignore the sites array passed in when the pin_map_id is invalid and always use site 0 internally, but I don't feel strongly about it. I think it's unlikely a client would ever do that.

**Review by jasonmreding — APPROVED** [review]

File: Source/Runtime/Clients/Session Management V1/Session Reservation/Get Connections Impl.vi
Comment (varshini-senthil) [reply Source/Runtime/Clients/Session Management V1/Session Reservation/Get Connections Impl.vi:1]: image

@jasonmreding, I would like to understand, if you are mentioning to remove the check for empty sites array and set the `pin map context.sites` value to [0], if the `pin map context.pin_map_id` value is an empty string?
- If so, I have added the additional check for empty sites array, to make sure we don't overwrite the value already set to `pin map context.sites` (considering the case of InstrumentStudio as client, where the value of sites array is internally set to [0], when the pin_map_id is empty)

image

Comment (jasonmreding) [reply Source/Runtime/Clients/Session Management V1/Session Reservation/Get Connections Impl.vi:1]: The original code I commented upon appeared like it would fail if a client passed an empty `pin_map_id` but specified a `sites` that wasn't empty and only contained site one for some reason. That's why I was suggesting you could only check for a valid `pin_map_id` rather than the `pin_map_id` is not empty *AND* the `sites` array is empty.

I don't understand the comment about overwriting the value already set to `pin map context.sites`. At this point in the code, it appears we are treating that as a read only field so I don't think there is any danger of overwriting anything. We already pass a modified sites array to down stream code.

Basically, what I'm suggesting is always ignore whatever the user passed in for sites when the pin_map_id is empty and make it just work regardless. However, as I said, I think the chances of a client ever doing this are pretty small so I don't feel strongly about it either.
